mixin/dashboards/defaults.libsonnet

```diff
@@ -13,7 +14,11 @@
   // Automatically add a uid to each dashboard based on the base64 encoding
   // of the file name and set the timezone to be 'default'.
   grafanaDashboards:: {
-    local component = std.split(filename, '.')[0],
+    local componentName = std.split(filename, '.')[0],
+    local componentParts = std.split(componentName, '_'),
```

Comment:
just a nit: We can have a `utils.sanitizeComponentName` to hide all the implementation details behind a function.